test/aspects/ec2-automated-shutdown/handler/index.test.ts

```diff
@@ -0,0 +1,222 @@
+                {
+                    Sns: {
+                        Message: JSON.stringify({
+                            version: '0',
+                            id: 'test-id',
+                            'detail-type': 'CloudWatch Alarm State Change',
+                            source: 'aws.cloudwatch',
+                            account: '123456789012',
+                            time: '2024-02-11T00:00:00Z',
+                            region: 'us-east-1',
+                            resources: [
+                                'arn:aws:cloudwatch:us-east-1:123456789012:alarm:LowCPUAlarm-1'
+                            ],
+                            detail: {
+                                alarmName: 'LowCPUAlarm-1',
+                                state: {
+                                    value: 'ALARM',
+                                    reason: 'Threshold Crossed',
+                                    reasonData: '{}',
+                                    timestamp: '2024-02-11T00:00:00Z'
+                                },
+                                previousState: {
+                                    value: 'OK',
+                                    reason: 'Threshold Crossed',
+                                    reasonData: '{}',
+                                    timestamp: '2024-02-11T00:00:00Z'
+                                },
+                                configuration: {
+                                    description:
+                                        'CPU utilization below threshold',
+                                    metrics: [
+                                        {
+                                            id: 'metric1',
+                                            metricStat: {
+                                                metric: {
+                                                    dimensions: [
+                                                        {
+                                                            name: 'InstanceId',
+                                                            value: 'i-1234567890abcdef0'
+                                                        }
+                                                    ]
+                                                }
+                                            }
+                                        }
+                                    ]
+                                }
+                            }
+                        })
+                    }
+                }
+            ]
+        };
+
+        // Mock the EC2 client constructor
+        (EC2Client as jest.Mock).mockImplementation(() => mockEC2Client);
+        mockEC2Client.send.mockReset();
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    it('should stop EC2 instance when alarm is in ALARM state', async () => {
+        // Arrange
+        const instanceId = 'i-1234567890abcdef0';
```

Comment:
We can store this mock instanceId in a fixtures folder for re-use since we are using it multiple times in this test